module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RecordsHelperTest.php

```diff
@@ -0,0 +1,339 @@
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithAjaxEnabled(): void
+    {
+        $config = new Config(['Authentication' => ['enableAjax' => true]]);
+        $loader = $this->createMock(Loader::class);
+
+        $ilsDetails1 = ['id' => 'record1', 'status' => 'available'];
+        $ilsDetails2 = ['id' => 'record2', 'status' => 'pending'];
+
+        $driver1 = $this->createMock(AbstractBase::class);
+        $driver2 = $this->createMock(AbstractBase::class);
+
+        $driver1->expects($this->once())
+            ->method('getExtraDetail')
+            ->with('ils_details')
+            ->willReturn($ilsDetails1);
+
+        $driver2->expects($this->once())
+            ->method('getExtraDetail')
+            ->with('ils_details')
+            ->willReturn($ilsDetails2);
+
+        $records = [$driver1, $driver2];
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats($records);
+
+        $this->assertIsArray($result);
+    }
+
+    /**
+     * Test collectRequestStats when ajax is disabled
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithAjaxDisabled(): void
+    {
+        $config = new Config(['Authentication' => ['enableAjax' => false]]);
+        $loader = $this->createMock(Loader::class);
+
+        $driver = $this->createMock(AbstractBase::class);
+        $driver->expects($this->never())
+            ->method('getExtraDetail');
+
+        $records = [$driver];
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats($records);
+
+        $this->assertNull($result);
+    }
+
+    /**
+     * Test collectRequestStats when enableAjax is not set (defaults to true)
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithDefaultAjaxSetting(): void
+    {
+        $config = new Config(['Authentication' => []]);
+        $loader = $this->createMock(Loader::class);
+
+        $ilsDetails = ['id' => 'record1', 'status' => 'available'];
+        $driver = $this->createMock(AbstractBase::class);
+
+        $driver->expects($this->once())
+            ->method('getExtraDetail')
+            ->with('ils_details')
+            ->willReturn($ilsDetails);
+
+        $records = [$driver];
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats($records);
+
+        $this->assertIsArray($result);
+    }
+
+    /**
+     * Test collectRequestStats with empty records array
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithEmptyRecords(): void
+    {
+        $config = new Config(['Authentication' => ['enableAjax' => true]]);
+        $loader = $this->createMock(Loader::class);
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats([]);
+
+        $this->assertIsArray($result);
```

Comment:
Could this be:

```suggestion
        $this->assertSame([], $result)
```

..?